packaging/setup/plugins/ovirt-engine-setup/ovirt-engine/config/jboss_fapolicyd.py

```diff
@@ -0,0 +1,88 @@
+#
+# ovirt-engine-setup -- ovirt engine setup
+#
+# Copyright oVirt Authors
+# SPDX-License-Identifier: Apache-2.0
+#
+#
+
+
+"""Jboss fapolicyd plugin."""
+
+
+import gettext
+import os
+
+from otopi import constants as otopicons
+from otopi import filetransaction
+from otopi import plugin
+from otopi import util
+
+
+from ovirt_engine import configfile
+from ovirt_engine import util as outil
+from ovirt_engine_setup import constants as osetupcons
+from ovirt_engine_setup.engine import constants as oenginecons
+from ovirt_engine_setup.engine_common import constants as oengcommcons
+
+
+def _(m):
+    return gettext.dgettext(message=m, domain='ovirt-engine-setup')
+
+
+@util.export
+class Plugin(plugin.PluginBase):
+    """JBoss fapolicyd plugin."""
+
+    def __init__(self, context):
+        super(Plugin, self).__init__(context=context)
+
+    @plugin.event(
+        stage=plugin.Stages.STAGE_INIT,
+    )
+    def _init(self):
+        self.environment.setdefault(
+            oengcommcons.FapolicydEnv.FAPOLICYD_ALLOW_OVIRT_RULE,
+            oengcommcons.FileLocations.FAPOLICYD_ALLOW_OVIRT_JBOSS_RULE
+        )
+
+    @plugin.event(
+        stage=plugin.Stages.STAGE_MISC,
+        condition=lambda self: self.environment[
+                                   oenginecons.CoreEnv.ENABLE
+                               ] and not self.environment[
+            osetupcons.CoreEnv.DEVELOPER_MODE
+        ],
+    )
+    def _misc(self):
+        config = configfile.ConfigFile([
+            oenginecons.FileLocations.OVIRT_ENGINE_SERVICE_CONFIG_DEFAULTS,
+            oenginecons.FileLocations.OVIRT_ENGINE_SERVICE_CONFIG
+        ])
+        engine_tmp_dir = os.path.join(
+            config.get('JBOSS_RUNTIME'),
+            'tmp'
+        )
+
+        self.environment[oengcommcons.FapolicydEnv.NEED_RESTART] = True
```

Comment:
Hmm, so that means we are going to put configuration file and restart fapolicyd for each engine-setup run. Can't we skip restarting fapolicyd, if configuration file already exist and it has correct content?

Reply:
Good point. Fapolicyd should not be restarted if the config is already there and even more it should not be started if was not running before seteup.

Reply:
I have a second thought about it - in very unlikely case when Wildfly/EAP runtime tmp directory location is changed, the modification will require fapolicyd restart. That means I am leaning towards always updating it and restarting anyway (that is a quick operation and can potentially save us during an upgrade). 
What I propose is only to check whether service has already been running to avoid accidentally starting it.

Reply:
After an offline discussion we have reached to conclusion that checking only the existence of the configuration is enough because Wildfly/EAP runtime tmp location is controlled internally and cannot be overridden by system administrators. To make things more explicit there we be a warning comment left in   55-allow-ovirt-jboss.in  where the rule is put